src/Microsoft.AspNetCore.Sockets.Client.Http/SendUtils.cs

```diff
@@ -119,5 +118,64 @@ protected override bool TryComputeLength(out long length)
                 return true;
             }
         }
+
+        private static class Log
+        {
+            private static readonly Action<ILogger, Exception> _sendStarted =
+                LoggerMessage.Define(LogLevel.Debug, new EventId(7, "SendStarted"), "Starting the send loop.");
+
+            private static readonly Action<ILogger, Exception> _sendStopped =
+                LoggerMessage.Define(LogLevel.Debug, new EventId(8, "SendStopped"), "Send loop stopped.");
+
+            private static readonly Action<ILogger, Exception> _sendCanceled =
+                LoggerMessage.Define(LogLevel.Debug, new EventId(9, "SendCanceled"), "Send loop canceled.");
+
+            private static readonly Action<ILogger, long, Uri, Exception> _sendingMessages =
+                LoggerMessage.Define<long, Uri>(LogLevel.Debug, new EventId(10, "SendingMessages"), "Sending {count} bytes to the server using url: {url}.");
+
+            private static readonly Action<ILogger, Exception> _sentSuccessfully =
+                LoggerMessage.Define(LogLevel.Debug, new EventId(11, "SentSuccessfully"), "Message(s) sent successfully.");
+
+            private static readonly Action<ILogger, Exception> _noMessages =
+                LoggerMessage.Define(LogLevel.Debug, new EventId(12, "NoMessages"), "No messages in batch to send.");
+
+            private static readonly Action<ILogger, Uri, Exception> _errorSending =
+                LoggerMessage.Define<Uri>(LogLevel.Error, new EventId(13, "ErrorSending"), "Error while sending to '{url}'.");
```

Comment:
Put a comment here similar to the ones in the other Log classes indicating that if you add a new EventID you need to reconcile it with the other loggers.

Reply:
Also, start them all at `100` so we don't have to deal with pushing event IDs around if we add a new one.